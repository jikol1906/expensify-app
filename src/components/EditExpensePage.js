import React from 'react';
import { connect } from 'react-redux';
import EditExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render(){
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <EditExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>Remove expense</button>
                </div>
            </div>
        )
    }

}

const mapStateToProps = (state , props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
};

const mapDispatchToProps = (dispatch) => {

    return {
        startEditExpense: (id,expense) => dispatch(startEditExpense(id,expense)),
        startRemoveExpense: (expenseId) => dispatch(startRemoveExpense(expenseId))
    }

};

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);