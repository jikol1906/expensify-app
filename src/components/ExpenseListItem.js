import React from 'react';
import { removeExpense } from "../actions/expenses";
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export default (props) => (
    <div>
        <NavLink to={"/edit/" + props.id}>
            <h3>{props.description}</h3>
        </NavLink>
        <p>
            {numeral(props.amount / 100).format('$0,0.00')}
            -
            {moment(props.createdAt).format('MMMM Do, YYYY')}
        </p>
    </div>
);





