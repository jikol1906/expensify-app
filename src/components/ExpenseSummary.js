import React from 'react';
import { Link } from 'react-router-dom';
import getVisibleExpenses from '../selectors/expenses';
import getExpenseTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpenseSummary =  ({expenseCount, expenseTotal}) => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{expenseCount}</span> expenses totalling: <span>{numeral( expenseTotal / 100).format('$0,0.00')}</span></h1>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
            </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {

    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

    return {
        expenseTotal: getExpenseTotal(visibleExpenses),
        expenseCount: visibleExpenses.length
    }

};

export default connect(mapStateToProps)(ExpenseSummary);
