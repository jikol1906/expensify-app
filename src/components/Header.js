import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/create" activeClassName="is-active" > Go to create page </NavLink>
        <NavLink to="/" activeClassName="is-active" exact={true}> Go to dashboard </NavLink>
    </header>
);

export default Header;