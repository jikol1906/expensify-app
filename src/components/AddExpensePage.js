import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from "../actions/expenses";
import Header from "./Header";

export class AddExpensePage extends React.Component {

    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };

    render() {

        return (
            <div>
                This is from my add expense component
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )

    }

}


const mapDispatchToProps = (dispatch) => {
    return {
        startAddExpense : (expense) => dispatch(startAddExpense(expense))
    }
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);