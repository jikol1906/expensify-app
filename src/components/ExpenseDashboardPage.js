import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';
import Header from "./Header";


const ExpenseDashboardPage = (props) => (
    <div>
        <ExpenseSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);


export default ExpenseDashboardPage