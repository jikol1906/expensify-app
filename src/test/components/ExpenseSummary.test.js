import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses';


test('should render ExpenseSummary correct with 1 expense', () => {

    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={2539} />);
    expect(wrapper).toMatchSnapshot();

});

test('should render ExpenseSummary correct with multiple expenses', () => {

    const wrapper = shallow(<ExpenseSummary expenseCount={23423} expenseTotal={25232342339} />);
    expect(wrapper).toMatchSnapshot();

});

