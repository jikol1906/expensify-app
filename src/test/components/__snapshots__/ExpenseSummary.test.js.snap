// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correct with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expenses totalling: 
    $25.39
  </h1>
</div>
`;

exports[`should render ExpenseSummary correct with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23423
     expenses totalling: 
    $252,323,423.39
  </h1>
</div>
`;
